hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/dynamicprotocol/DynamicProtocolService.java

```diff
@@ -51,29 +55,29 @@ public static Builder builder() {
 
     private final Map<String, DynamicEndpointService> endpointServices;
     private ScheduledExecutorService threadPool;
+    private final Map<String, ScheduledFuture<?>> jobs;
 
     public DynamicProtocolService(Map<String, DynamicEndpointService> endpointServices) {
         this.endpointServices = endpointServices;
+        this.jobs = new HashMap<>();
     }
 
+    /**
+     * @return a shallow copy of all endpoint services
+     */
     public Map<String, DynamicEndpointService> getDynamicEndpointServices() {
-        return endpointServices;
+        synchronized (endpointServices) {
+            return new HashMap<>(endpointServices);
+        }
     }
 
     public void start() {
         ThreadFactory threadFactory = ThreadFactoryGenerator.generateFactory(true,
                 "Hawkular WildFly Agent Dynamic Protocol Service Thread Pool");
         threadPool = Executors.newScheduledThreadPool(1, threadFactory);
```

Comment:
governed by the lifecycle of the agent service - this start is called only when wildfly starts the service. same with stop (when wildfly shuts down the service).